### PR TITLE
feat (ui): user friendly create new project

### PR DESCRIPTION
**Pull request by yesnault**
Signed-off-by: Yvonnick Esnault <yvonnick.esnault@corp.ovh.com>

### CODE REVIEW

**Comment by yesnault** [comment]
createproject

File: engine/api/project.go [hunk 1]
```diff
@@ -4,20 +4,21 @@ import (
...
 	"github.com/ovh/cds/engine/api/group"
+	"github.com/ovh/cds/engine/api/permission"
 	"github.com/ovh/cds/engine/api/project"
 	"github.com/ovh/cds/sdk"
 	"github.com/ovh/cds/sdk/log"
 )
 
 func getProjectsHandler(w http.ResponseWriter, r *http.Request, db *gorp.DbMap, c *businesscontext.Ctx) error {
-	// Get project name in URL
 	withApplication := FormBool(r, "application")
```
Comment (fsamin) [inline engine/api/project.go]: why deleting comments ?

File: sdk/project.go [hunk 2]
```diff
@@ -9,6 +9,8 @@ import (
 	"time"
 )
 
+const DefaultGroupOnProjectSuffix = "_PRJ"
```
Comment (fsamin) [inline sdk/project.go]: sorry i don't like that :/

File: engine/api/group.go [hunk 3]
```diff
@@ -129,6 +130,10 @@ func updateGroupHandler(w http.ResponseWriter, r *http.Request, db *gorp.DbMap,
 	}
 	defer tx.Rollback()
 
+	if strings.HasSuffix(updatedGroup.Name, sdk.DefaultGroupOnProjectSuffix) {
+		return sdk.WrapError(sdk.ErrWrongRequest, "You can create a group with ending with %s", sdk.DefaultGroupOnProjectSuffix)
```
Comment (fsamin) [inline engine/api/group.go]: i don't understand, if the group ends with the PRJ suffix, we return an error telling him he can do it ?

File: engine/api/group.go [hunk 4]
```diff
@@ -210,6 +215,9 @@ func addGroupHandler(w http.ResponseWriter, r *http.Request, db *gorp.DbMap, c *
 	}
 	defer tx.Rollback()
 
+	if strings.HasSuffix(g.Name, sdk.DefaultGroupOnProjectSuffix) {
+		return sdk.WrapError(sdk.ErrWrongRequest, "You can create a group with ending with %s", sdk.DefaultGroupOnProjectSuffix)
```
Comment (fsamin) [inline engine/api/group.go]: i don't understand, if the group ends with the PRJ suffix, we return an error telling him he can do it ?

File: engine/api/project.go [hunk 5]
```diff
@@ -157,9 +156,29 @@ func addProjectHandler(w http.ResponseWriter, r *http.Request, db *gorp.DbMap, c
...
+	for i := range p.ProjectGroups {
+		groupPermission := &p.ProjectGroups[i]
+		if strings.TrimSpace(groupPermission.Group.Name) == "" {
+			continue
+		}
+		groupAttached = true
+	}
+	if !groupAttached {
+		permG := sdk.GroupPermission{
+			Group:      sdk.Group{Name: p.Name + sdk.DefaultGroupOnProjectSuffix},
```
Comment (fsamin) [inline engine/api/project.go]: i'm not sure, i am happy with this.
If you want to make project permission easier to manager in CDS, creating a group for each project won't solve this problem. 
BTW i admit that we have something to do about this

File: ui/src/assets/i18n/en.json [hunk 6]
```diff
@@ -338,6 +338,8 @@
   "project_advanced_title" : "Project administration",
   "project_added" : "Project has just been created",
   "project_add_ssh_key" : "Generate a ssh key",
+  "project_add_ssh_key_help" : "A SSK Key is useful when you want to git clone some code from a Git Repository. CDS can generate a key pair and you can copy / paste the public key to your access key on your repositories.",
```
Comment (fsamin) [inline ui/src/assets/i18n/en.json]: to your access keys

File: ui/src/assets/i18n/en.json [hunk 7]
```diff
@@ -338,6 +338,8 @@
...
+  "project_add_group_help": "Read / Write / Execute permissions have to be setted to this new project. You can choose an existing user's group or let empty to automatically create a new group, you will be administrator of this group.",
```
Comment (fsamin) [inline ui/src/assets/i18n/en.json]: to set/set/set
to set on
users group

File: engine/api/project.go (hunk 1 above)
Comment (yesnault) [reply engine/api/project.go]: Because it's false :) We don't get project name in URL on this handler

File: engine/api/project.go (hunk 5 above)
Comment (yesnault) [reply engine/api/project.go]: The group is automatically created only if user don't select an existing group. But yes, it's only a suggestion

File: sdk/project.go (hunk 2 above)
Comment (yesnault) [reply sdk/project.go]: yes, me too.

File: engine/api/project.go (hunk 5 above)
Comment (yesnault) [reply engine/api/project.go]: In fact, there two kinds of users:
- newcomers: they use the same group name as project name. Today, they often think that the group is only usable on this project (group name like "admins" for example)
- users familiar with CDS: they probably want to choose an existing group, so they can still choose the group.

Comment (fsamin) [reply engine/api/project.go]: yup, the permissions are way too complicated to newcomers.
They should be able to work on their first pipeline after 2 clicks on UI

File: engine/api/project.go (hunk 1 above)
Comment (fsamin) [reply engine/api/project.go]: indeed
or fix it :)

Comment (yesnault) [reply engine/api/project.go]: other comments are back

**Comment by yesnault** [comment]
So, no more groupe with suffix "_PRJ" created. The “auto-group" is created with the name of project, less weird.

File: ui/src/app/views/project/add/project.add.component.ts [hunk 8]
```diff
@@ -79,21 +79,19 @@ export class ProjectAddComponent {
                 this.keyError = true;
             }
         }
-        if (!this.group || this.group.name === '') {
-            this.groupError = true;
+        if (this.group || this.group.name !== '') {
```
Comment (sguiheux) [inline ui/src/app/views/project/add/project.add.component.ts]: s/||/&&/g

File: ui/src/app/views/project/add/project.add.html [hunk 9]
```diff
@@ -48,6 +36,30 @@
...
+                <div class="field">
+                  {{ 'project_permission_form_wizard_title' | translate }} <a class="pointing" (click)="modalCreateGroup.show()">{{ 'project_permission_form_wizard_new' | translate }}</a>
+                  <sm-select class="fluid search"
+                      [options]="{'fullTextSearch': true}"
+                      [model]="group?.id"
+                      (modelChange)="setGroup($event)"
+                      id="PermissionGroup">
+                      <option *ngFor="let g of groupList" [value]="g.id">{{g.name}}</option>
+                  </sm-select>
+                  <div class="row" *ngIf="groupError">
```
Comment (sguiheux) [inline ui/src/app/views/project/add/project.add.html]: groupError still here?

File: ui/src/app/views/project/add/project.add.component.ts [hunk 10]
```diff
@@ -79,21 +79,19 @@ export class ProjectAddComponent {
                 this.keyError = true;
             }
         }
-        if (!this.group || this.group.name === '') {
-            this.groupError = true;
```
Comment (sguiheux) [inline ui/src/app/views/project/add/project.add.component.ts]: remove from file

File: ui/src/assets/i18n/fr.json [hunk 11]
```diff
@@ -338,6 +338,8 @@
   "project_advanced_title" : "Administration du projet",
   "project_added" : "Projet créé",
   "project_add_ssh_key" : "Générer une clé ssh",
+  "project_add_ssh_key_help" : "Une clé ssh est généralement utile lorsque vous souhaitez cloner un dépôt Git. CDS peut générer une paire de clé, il ne vous restera qu'à copier / coller la clé publique fournie par CDS dans les clés autorisées sur vos dépôt.",
```
Comment (sguiheux) [inline ui/src/assets/i18n/fr.json]: dépôts

File: ui/src/assets/i18n/fr.json [hunk 12]
```diff
@@ -338,6 +338,8 @@
...
+  "project_add_group_help": "Les persmission de Lecture / Écriture / Exécution doivent être positionnées sur ce nouveau projet. Vous pouvez choisir un groupe d'utilisateurs existant ou laisser vide pour que CDS créé automatiquement un nouveau groupe, vous serez administrateur de ce groupe.",
```
Comment (sguiheux) [inline ui/src/assets/i18n/fr.json]: permissions

Comment (sguiheux) [reply ui/src/assets/i18n/fr.json]: , vous en serez l'administrateur

**Pull request merged**